### PR TITLE
DIT-1106 Complete a Case

### PR DESCRIPTION
**Pull request by ed0906**

### CODE REVIEW

File: app/uk/gov/hmrc/tariffclassificationfrontend/models/Permission.scala [hunk 1]
```diff
@@ -23,20 +23,20 @@ object Permission extends Enumeration {
...
   val readOnlyPermissions: Set[Permission] = Set(SEARCH, ADVANCED_SEARCH, VIEW_CASES)
 
   val systemAdminPermissions: Set[Permission] = Set(MANAGE_KEYWORDS, MANAGE_QUEUES)
 
   private val managerPermissions: Set[Permission] = Set(CREATE_CASES, VIEW_MY_CASES, VIEW_QUEUE_CASES, VIEW_ASSIGNED_CASES, SEARCH, ADVANCED_SEARCH, VIEW_CASES, ASSIGN_CASE, VIEW_CASE_ASSIGNEE,
     RELEASE_CASE, SUPPRESS_CASE, REFER_CASE, REOPEN_CASE, REJECT_CASE, SUSPEND_CASE, COMPLETE_CASE, REMOVE_ATTACHMENTS,
-    CANCEL_CASE, ADD_NOTE, ADD_ATTACHMENT, KEYWORDS, EDIT_RULING, APPEAL_CASE, EXTENDED_USE, MOVE_CASE_BACK_TO_QUEUE, VIEW_REPORTS, EDIT_SAMPLE)
+    CANCEL_CASE, ADD_NOTE, ADD_ATTACHMENT, KEYWORDS, EDIT_LIABILITY, EDIT_RULING, APPEAL_CASE, EXTENDED_USE, MOVE_CASE_BACK_TO_QUEUE, VIEW_REPORTS, EDIT_SAMPLE)
 
   private val teamBasicPermissions: Set[Permission] = Set(CREATE_CASES, VIEW_MY_CASES, VIEW_QUEUE_CASES, SEARCH, ADVANCED_SEARCH, VIEW_CASES, ASSIGN_CASE, VIEW_CASE_ASSIGNEE,
```
Comment (pllaurado) [inline app/uk/gov/hmrc/tariffclassificationfrontend/models/Permission.scala]: There is an special case for this edit liability. If the case is NEW anyone (OWNER, NOT-OWNER, ADMIN) can edit the liability.

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/confirm_complete_case.scala.html [hunk 2]
```diff
@@ -18,24 +18,30 @@
...
                 <p>The ruling has been published.</p>
                 <p>View the published ruling on the public rulings website.</p>
                 <p>The applicant has been notified by email.</p>
                 <p class="bold">The status of this case is now completed</p>
+            } else {
+                <h2 class="heading-large mt-0">This liability decision is complete</h2>
 
-                @confirmation_next_steps()
-            </div>
+                <p>Email the applicant with a copy of tthis decision.</p>
```
Comment (aaroncarroll82) [inline app/uk/gov/hmrc/tariffclassificationfrontend/views/confirm_complete_case.scala.html]: Think Aimee pointed out this typo on 'tthis' - did you fix it?

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/liability_details.scala.html [hunk 3]
```diff
@@ -47,11 +47,13 @@
 
             <h3 class="heading-medium mt-0">@messages("case.liability.about.title") </h3>
 
-            <div id="edit-liability-details" class="grid-row  mb-2">
-                <span class="column-one-third">
-                    <a href="@routes.LiabilityController.editLiabilityDetails(c.reference)"> Edit details </a>
-                </span>
-            </div>
+            @if(c.status == CaseStatus.OPEN || c.status == CaseStatus.NEW && request.hasPermission(Permission.EDIT_LIABILITY)) {
```
Comment (aaroncarroll82) [inline app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/liability_details.scala.html]: Can these details not be edited when it's in referred status?

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/confirm_complete_case.scala.html (hunk 2 above)
Comment (ed0906) [reply app/uk/gov/hmrc/tariffclassificationfrontend/views/confirm_complete_case.scala.html]: Sorted, cheers

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/liability_details.scala.html (hunk 3 above)
Comment (ed0906) [reply app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/liability_details.scala.html]: I think referred usually mean "unactionable" so I would presume no

File: app/uk/gov/hmrc/tariffclassificationfrontend/models/Permission.scala (hunk 1 above)
Comment (ed0906) [reply app/uk/gov/hmrc/tariffclassificationfrontend/models/Permission.scala]: I'll relax the  controller validation so it just checks the case status. I'll implement the proper rule on the edit button

Comment (ed0906) [reply app/uk/gov/hmrc/tariffclassificationfrontend/models/Permission.scala]: Done

Comment (pllaurado) [reply app/uk/gov/hmrc/tariffclassificationfrontend/models/Permission.scala]: Looks good!

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/liability_details.scala.html (hunk 3 above)
Comment (pllaurado) [reply app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/liability_details.scala.html]: exactly, when we refer a case then we wait for action ( unattended ) to be committed meanwhile the case is on hold as application perspective.  Once this action is finished the same operator will reopen the case.

**Comment by aaroncarroll82** [comment]
Don't know if it was part of this ticket or not (believe it might be), but the 'you must fill these fields in' hint that appears on bottom ruling section lists commodity code twice if you don't fill it in.
Screenshot 2019-06-13 at 13 36 45

**Comment by ed0906** [comment]
> Don't know if it was part of this ticket or not (believe it might be), but the 'you must fill these fields in' hint that appears on bottom ruling section lists commodity code twice if you don't fill it in.
> Screenshot 2019-06-13 at 13 36 45

I've mapped the error codes to a set which should sort it. It's because theres multiple constraints on com codes. Good spot

**Pull request merged**